perimeter/shared_access.sp

```diff
@@ -895,3 +911,761 @@ control "rds_db_snapshot_shared_with_trusted_accounts" {
+}
+
+control "s3_bucket_policy_prohibits_untrusted_organizations_access" {
+  title       = "S3 bucket policies should prohibit untrusted organization access"
+  description = "Check if S3 bucket policies allows access to untrusted organizations"
+  sql         = replace(replace(local.resource_policy_shared_organizations_sql_region, "__TABLE_NAME__", "aws_s3_bucket"), "__ARN_COLUMN__", "arn")
+
+  param "trusted_organizations" {
+    description = "A list of trusted organizations."
+    default     = var.trusted_organizations
+  }
+
+  param "trusted_organization_units" {
+    description = "A list of trusted organizations unit."
+    default     = var.trusted_organization_units
+  }
+
+  tags = merge(local.aws_perimeter_common_tags, {
+    service = "AWS/S3"
+  })
+}
+
+control "sns_topic_policy_prohibits_untrusted_organizations_access" {
+  title       = "SNS topic policies should prohibit untrusted organization access"
+  description = "Check if SNS topic policies allows access to untrusted organizations"
+  sql         = replace(replace(local.resource_policy_shared_organizations_sql_region, "__TABLE_NAME__", "aws_sns_topic"), "__ARN_COLUMN__", "topic_arn")
+
+  param "trusted_organizations" {
+    description = "A list of trusted organizations."
+    default     = var.trusted_organizations
+  }
+
+  param "trusted_organization_units" {
+    description = "A list of trusted organizations unit."
+    default     = var.trusted_organization_units
+  }
+
+  tags = merge(local.aws_perimeter_common_tags, {
+    service = "AWS/SNS"
+  })
+}
+
+control "sqs_queue_policy_prohibits_untrusted_organizations_access" {
+  title       = "SQS queue policies should prohibit untrusted organization access"
+  description = "Check if SQS queue policies allows access to untrusted organizations"
+  sql         = replace(replace(local.resource_policy_shared_organizations_sql_region, "__TABLE_NAME__", "aws_sqs_queue"), "__ARN_COLUMN__", "queue_arn")
+
+  param "trusted_organizations" {
+    description = "A list of trusted organizations."
+    default     = var.trusted_organizations
+  }
+
+  param "trusted_organization_units" {
+    description = "A list of trusted organizations unit."
+    default     = var.trusted_organization_units
+  }
+
+  tags = merge(local.aws_perimeter_common_tags, {
+    service = "AWS/SQS"
+  })
+}
+
+locals {
+  resource_policy_shared_services_sql = <<-EOT
+    select
+      r.__ARN_COLUMN__ as resource,
+      case
+        when jsonb_array_length(pa.allowed_principal_services) = 0 then 'ok'
+        when pa.access_level = 'public' then 'alarm'
+        when pa.allowed_principal_services <@ to_jsonb(($1)::text[]) then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when jsonb_array_length(pa.allowed_principal_services) = 0 then title || ' trust policy does not reference any services.'
+        when pa.access_level = 'public' then title || ' trust policy allows service access to a large range of untrusted accounts.'
```

Comment:
Similar to the organization query comment, `a large range` is confusing. Also, should it be `untrusted services` instead of accounts, or something more specific like `any service from all accounts`?